roxie/udplib/udptrs.cpp

```diff
@@ -571,8 +571,9 @@ class UdpReceiverEntry : public IUdpReceiverEntry
                         const char *data = buffer->data + sizeof(UdpPacketHeader);
                         const MemoryAttr &udpkey = getSecretUdpKey(true);
                         aesEncrypt(udpkey.get(), udpkey.length(), data, length, encryptBuffer);
-                        header->length = encryptBuffer.length();
-                        encryptBuffer.writeDirect(0, sizeof(UdpPacketHeader), header);   // Only really need length updating
+                        UdpPacketHeader newHeader;
+                        newHeader.length = encryptBuffer.length();
+                        encryptBuffer.writeDirect(offsetof(UdpPacketHeader, length), sizeof(newHeader.length), &newHeader.length);  // Only need to update the length - rest is the same
```

Comment:
ok, I agree, changing header->length does not seem right, should be a copy.  Good find.
But why use newHeader, why not a u_short ?  Is it because you want to keep the abstraction of the internal details of the packet ?

Reply:
It was to avoid explicitly encoding the type of this field.  Do you know if there is a c++ standard equivalent of typeof?

Reply:
typeid ???